src/test/java/org/jenkinsci/plugins/workflow/graphanalysis/ForkScannerTest.java

```diff
@@ -1092,11 +1093,12 @@ public void parallelBranchOrdering() throws Exception {
     private static void assertBranchOrder(FlowExecution execution, String... expectedBranchNames) {
         ForkScanner scanner = new ForkScanner();
         scanner.setup(execution.getCurrentHeads());
-        List<String> branches = StreamSupport.stream(scanner.spliterator(), false)
+        String[] branches = StreamSupport.stream(scanner.spliterator(), false)
                 .map(n -> n.getPersistentAction(ThreadNameAction.class))
                 .filter(Objects::nonNull)
                 .map(ThreadNameAction::getThreadName)
-                .collect(Collectors.toList());
-        assertThat(branches, contains(expectedBranchNames));
+                .collect(Collectors.toList())
+                .toArray(String[]::new);
+        assertThat(branches, equalTo(expectedBranchNames));
```

Comment:
If it does fail again, this allows us to see the full order in the test failure message.

Reply:
Do you mean to use `arrayEqualTo` or similar? I do not think `equalTo` will work on arrays generally.

Reply:
There is a special case for arrays, see https://github.com/hamcrest/JavaHamcrest/blob/e4b9afdc2624bbebef3c7c739e6074ecd8a6c06e/hamcrest/src/main/java/org/hamcrest/core/IsEqual.java#L36-L38, which ends up being delegated to by matchers like `arrayContaining` (`arrayEqualTo` is not a thing as far as I can see). I can switch to `arrayContaining` though if you prefer.

Maybe you are thinking of `junit`'s `Assert.assertEquals` vs `Assert.assertArrayEquals`?